core/src/main/java/org/infinispan/configuration/cache/IndexingConfigurationBuilder.java

```diff
@@ -204,6 +211,14 @@ public void validate() {
          }
       }
 
+      if (attributes.attribute(PROPERTIES).get()
+                    .entrySet().stream()
+                    .filter(e -> ((String) e.getKey()).endsWith(DIRECTORY_PROVIDER_SUFFIX))
+                    .map(Map.Entry::getValue)
+                    .collect(Collectors.toSet()).size() > 1) {
+         throw CONFIG.foundMultipleDirectoryProviders();
```

Comment:
You need to take into account that the default is ```fs```, and can be omitted. So having 2 entities with one of them pointing to ```local-heap```, would cause only 1  property with DIRECTORY_PROVIDER_SUFFIX by two backends.

Reply:
Good find. I will refine the validation to take defaults into account

Reply:
Fixed